app/src/main/java/io/homeassistant/companion/android/nfc/NfcEditFragment.kt

```diff
@@ -0,0 +1,110 @@
+import android.view.View
+import android.view.ViewGroup
+import android.widget.Toast
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.navigation.fragment.findNavController
+import io.homeassistant.companion.android.R
+import io.homeassistant.companion.android.common.dagger.GraphComponentAccessor
+import io.homeassistant.companion.android.domain.integration.IntegrationUseCase
+import javax.inject.Inject
+import kotlinx.android.synthetic.main.fragment_nfc_edit.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.launch
+
+/**
+ * A simple [Fragment] subclass as the second destination in the navigation.
+ */
+class NfcEditFragment : Fragment() {
+
+    val TAG = NfcEditFragment::class.simpleName
+
+    private val mainScope: CoroutineScope = CoroutineScope(Dispatchers.Main + Job())
+
+    private lateinit var viewModel: NfcViewModel
+
+    @Inject
+    lateinit var integrationUseCase: IntegrationUseCase
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        viewModel = ViewModelProvider(requireActivity()).get(NfcViewModel::class.java)
+
+        // Inject components
+        DaggerProviderComponent
+            .builder()
+            .appComponent((activity?.application as GraphComponentAccessor).appComponent)
+            .build()
+            .inject(this)
+
+        // Inflate the layout for this fragment
+        return inflater.inflate(R.layout.fragment_nfc_edit, container, false)
+    }
+
+    @SuppressLint("SetTextI18n")
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        val nfcReadObserver = Observer<String> { uuid ->
+            mainScope.launch {
+                et_tag_identifier_content.setText(uuid)
+                val deviceName = integrationUseCase.getRegistration().deviceName!!
+                et_tag_example_trigger_content.setText("- platform: event\n  event_type: nfc.tag_read\n  event_data:\n    device_name: $deviceName\n    tag: $uuid")
+            }
+        }
+        viewModel.nfcReadEvent.observe(viewLifecycleOwner, nfcReadObserver)
+
+        btn_tag_duplicate.setOnClickListener {
+            viewModel.nfcWriteTagEvent.postValue(et_tag_identifier_content.text.toString())
+            findNavController().navigate(R.id.action_NFC_WRITE)
+        }
+
+        btn_tag_fire_event.setOnClickListener {
+            mainScope.launch {
+                val uuid: String = viewModel.nfcReadEvent.value.toString()
+                val deviceName = integrationUseCase.getRegistration().deviceName!!
+                try {
+                    integrationUseCase.fireEvent(
+                        "nfc.tag_read",
```

Comment:
I made a design doc for the tag integration the other day, [it's here](https://docs.google.com/document/d/1b33WrKvNUFqxB-Q7yv4g05uZ_4v7IS3lQY7yDgSUnL4/edit), I suggest we add a new webhook `tag_scanned` to the mobile app and let it work with that. I will open a PR now so it's in the beta/next release.

Reply:
Update: realized we shouldn't do anything until the tag integration is added, or else we need to make a breaking change.

Reply:
Ok, made PR https://github.com/home-assistant/core/pull/38721, we can do it without a breaking change.


Send webhook type `scan_tag` with data `{"tag_id": "abcd"}`

Reply:
@balloob Thank you! How do I call the webhook? 🤔 Is the following correct?

`integrationUseCase.fireEvent("scan_tag", hashMapOf("tag_id" to uuid))`

Reply:
You'll need to add a new function for it all the way down the stack (in IntegrationRepositoryImpl).

Reply:
Oh okay, I see, thank you!. So I added the new function. Just to double check.. this logcat output looks correct, right?

``` 
--> POST https://home-assistant.xxx.com/api/webhook/bcc461...
D/OkHttp: Content-Type: application/json; charset=UTF-8
D/OkHttp: Content-Length: 76
D/OkHttp: {"type":"scan_tag","data":{"tag_id":"123e4567-e89b-12d3-a456-426614174000"}}
```

Reply:
Yep that looks good!  We'll also want to make sure we only do that for Core versions 114 and above since that's the first version that has the new webhook.  @balloob also suggested we fire the event for versions below 114.

Reply:
@JBassett Sorry I'm a little confused by your sentence. Do I need to make a check or are we just firing the event anyways? It's no harm to send it if it's not supported, right?